src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -201,48 +219,47 @@ export class EthereumBlockchainService implements BlockchainService {
    *   the gas price we set by a 10% multiple with each subsequent attempt
    * @private
    */
-  async setGasPrice(txData: TransactionRequest, attempt: number): Promise<void> {
+  async setGasPrice(txData: TransactionRequest): Promise<void> {
     if (this.overrideGasConfig) {
-      txData.gasLimit = BigNumber.from(this.gasLimit)
+      txData.gasLimit = this.gasLimit
       logger.debug('Overriding Gas limit: ' + txData.gasLimit.toString())
       return
     }
 
-    const feeData = await this.wallet.provider.getFeeData()
+    const walletProvider = Utils.coalesce(() => this.wallet.provider, 'No wallet provider')
+    const feeData = await Utils.coalesceAsync(() => walletProvider.getFeeData(), 'No fee data')
     // Add extra to gas price for each subsequent attempt
     const maxFeePerGas = feeData.maxFeePerGas
     const maxPriorityFeePerGas = feeData.maxPriorityFeePerGas
     // Is EIP-1559
     if (maxPriorityFeePerGas && maxFeePerGas) {
-      // When attempt 0, use currently estimated maxPriorityFeePerGas; otherwise use previous transaction maxPriorityFeePerGas
-      const prevPriorityFee = BigNumber.from(
-        txData.maxPriorityFeePerGas || feeData.maxPriorityFeePerGas
-      )
+      const baseFee = maxFeePerGas - maxPriorityFeePerGas
+      const prevPriorityFee = BigInt(txData.maxPriorityFeePerGas || feeData.maxPriorityFeePerGas)
       const nextPriorityFee = EthereumBlockchainService.increaseGasPricePerAttempt(
         maxPriorityFeePerGas,
-        attempt,
         prevPriorityFee
       )
       txData.maxPriorityFeePerGas = nextPriorityFee
-      const baseFee = maxFeePerGas.sub(maxPriorityFeePerGas)
-      txData.maxFeePerGas = baseFee.add(nextPriorityFee)
+      // must be greater than baseFee + maxPriorityFeePerGas
+      txData.maxFeePerGas = addPadding(baseFee + nextPriorityFee)
```

Comment:
why are we adding padding here if we already added padding in `increaseGasPricePerAttempt`?  Is this redundant?

Reply:
On upgrading ethers, we run into underpriced transaction errors. I'm not sure if ethers changed something in the data they report back, but this is required to not have underpriced transaction errors.